test/extended/cli/builds.go

```diff
@@ -0,0 +1,179 @@
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	"k8s.io/kubectl/pkg/util/slice"
+)
+
+const (
+	bcOutputTemplate      = "{{with .spec.output.to}}{{.kind}} {{.name}}{{end}}"
+	testDockerFileContent = "FROM image-registry.openshift-image-registry.svc:5000/openshift/tests:latest"
+)
+
+type expectedBuild struct {
+	bcName      string
+	isName      string
+	sourceType  string
+	outputType  string
+	outputImage string
+}
+
+func (r *expectedBuild) setDefaults() {
+	if r.outputType == "ImageStreamTag" && r.isName == "" {
+		r.isName = r.bcName
+	}
+}
+
+func testNewBuild(oc *exutil.CLI, opts expectedBuild) {
+	opts.setDefaults()
+	// new build
+	buildOut, err := oc.Output()
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(buildOut).To(o.And(
```

Comment:
Shouldn't we wait for the build? Isn't the whole purpose of moving to go code is to be reasonable with the time frame and have knobs to control it?

Reply:
Yes, I could add it. Please consider also https://github.com/openshift/origin/pull/26141#issuecomment-836731670